### PR TITLE
Add Coupon BxGy restriction

### PR DESCRIPTION
**Pull request by mikerebilly**
Add Coupon Restriction for BxGy -- Buy X, Get Y.

### CODE REVIEW

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                - $ref: ../../ResourceId.yaml
+            quantity:
+              type: integer
+              minimum: 1
+              description: Total quantity of the associated plan ID that must be purchased for this restriction to apply.
+      get:
+        type: array
+        description: >
+          One or more plan and quantity definitions which will be added to a purchase if all of the definitions in `buy` are met.
+          All of these will be applied.
```
Comment (VladimirBerdnik) [inline openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:34]:
```suggestion
          One or more plan and quantity definitions which will be added to a purchase for free if all of the definitions in `buy` are met.
```

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  - type: object
+    required:
+      - buy
+      - get
+    properties:
+      buy:
+        type: array
+        description: >
+          One or more plan and quantity definitions to qualify for receiving all plans and quantity defined in `get`.
+          All of these pairs must be met for the `get` definition(s) to apply.
```
Comment (VladimirBerdnik) [inline openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:14]:
```suggestion
          One or more plan and quantity definitions that must be purchased for receiving all plans and quantity defined in `get`.
```

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml (hunk 1 above)
Comment (VladimirBerdnik) [reply openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:34]: well, probably not `for free`

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml (hunk 2 above)
Comment (mikerebilly) [reply openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:14]: So for both of these suggestions, you're basically saying that the second phrase is unnecessary? (Both suggestions remove the "all of these will be applied" or "all of these must be met").

Comment (mikerebilly) [reply openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml]: Ok yeah i see that, it's kind of built into the first sentences. Ok I've pushed changes.

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  - $ref: ../CouponRestriction.yaml
+  - $ref: ../RedemptionRestriction.yaml
+  - type: object
+    required:
+      - buy
+      - get
+    properties:
+      buy:
+        type: array
+        description: One or more plan and quantity definitions that must be purchased for receiving all plans and quantity defined in `get`.
```
Comment (noon-dawg) [inline openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:12]: nitpick: sentence is difficult to follow
```suggestion
        description: One or more plan and quantity definitions that must be purchased to receive all plans and quantities that are defined in `get`.
```

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+              description: Plan ID of what is being purchased.
+              allOf:
+                - $ref: ../../ResourceId.yaml
+            quantity:
+              type: integer
+              minimum: 1
+              description: Total quantity of the associated plan ID that must be purchased for this restriction to apply.
+      get:
+        type: array
+        description: One or more plan and quantity definitions which will be added to a purchase if all of the definitions in `buy` are met.
```
Comment (noon-dawg) [inline openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:30]:
```suggestion
        description: One or more plan and quantity definitions which will be added to a purchase if all of the definitions in the `buy` operation are met.
```

**Review by rensftw — APPROVED** [review]

**Review by VladimirBerdnik — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Review by XaosSintez — APPROVED** [review]

**Pull request merged**